channeldb/payments_test.go

```diff
@@ -0,0 +1,163 @@
+package channeldb
+
+import (
+	"testing"
+	"time"
+	"github.com/roasbeef/btcutil"
+	"bytes"
+	"reflect"
+	"github.com/davecgh/go-spew/spew"
+	"math/rand"
+	"fmt"
+	"github.com/btcsuite/fastsha256"
+)
+
+func makeFakePayment() *OutgoingPayment {
+	// Create a fake invoice which we'll use several times in the tests
+	// below.
+	fakeInvoice := &Invoice{
+		CreationDate: time.Now(),
+	}
+	fakeInvoice.Memo = []byte("memo")
+	fakeInvoice.Receipt = []byte("recipt")
+	copy(fakeInvoice.Terms.PaymentPreimage[:], rev[:])
+	fakeInvoice.Terms.Value = btcutil.Amount(10000)
+	// Make fake path
+	fakePath := make([][]byte, 3)
+	for i:=0; i<3; i++ {
+		fakePath[i] = make([]byte, 33)
+		for j:=0; j<33; j++ {
+			fakePath[i][j] = byte(i)
+		}
+	}
+	var rHash [32]byte = fastsha256.Sum256(rev[:])
+	fakePayment := & OutgoingPayment{
+		Invoice: *fakeInvoice,
+		Fee: 101,
+		Path: fakePath,
+		TimeLockLength: 1000,
+		RHash: rHash,
+	}
+	return fakePayment
+}
+
+// randomBytes creates random []byte with length
+// in range [minLen, maxLen)
+func randomBytes(minLen, maxLen int) []byte {
+	l := minLen + rand.Intn(maxLen-minLen)
+	b := make([]byte, l)
+	_, err := rand.Read(b)
+	if err != nil {
+		panic(fmt.Sprintf("Internal error. Cannot generate random string: %v", err))
```

Comment:
Using a panic here would cause _all_ other tests to halt (if they were using `t.Errorf` statements rather than `t.Fatalf` statements).

Instead of using `panic, I think we have two alternatives: 
1. Pass in a `t.Testing` and fail the test upon an error from the rand read
2. Simply return an error here and handle it properly at the calling site

I'll let it up to you as to which option you'd like to pursue.